core/io/file_access.cpp

```diff
@@ -107,6 +111,10 @@ Ref<FileAccess> FileAccess::open(const String &p_path, int p_mode_flags, Error *
 	if (r_error) {
 		*r_error = err;
 	}
+	if (ret.is_null()) {
+		// We need to agree on whether ret.is_null() means 'file cannot be opened' or 'file not found'
+		*r_error = ERR_FILE_NOT_FOUND;
+	}
```

Comment:
```suggestion
```
Same here, this can only be reached if `ret->open_internal` above can be called, if `ret.is_null()` is `true` there will be a segmentation fault above.